### ISSUE TITLE
Cannot run text-generation example using wgpu

### ISSUE THREAD

**Issue by tafia** [tracel-ai/burn#3217]
**Describe the bug**

I am trying to use wgpu to run the text-generation example. But I get some autotune related errors:

```
thread 'main' panicked at /home/johann/projects/burn/crates/burn-cubecl-fusion/src/shared/trace/base.rs:94:38:
Tensors are not approx eq:
  => Position 0: 0.00000017888068 != 0.00000008146409
     diff (rel = +5.45e-1, abs = +9.74e-8), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 1: 0.00000024821009 != 0.0000002456496
     diff (rel = +1.03e-2, abs = +2.56e-9), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 2: 0.0000004017506 != 0.00000049207665
     diff (rel = +1.84e-1, abs = +9.03e-8), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 3: 0.00000023997717 != 0.00000014132979
     diff (rel = +4.11e-1, abs = +9.86e-8), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 4: 0.00000017888068 != 0.00000015982457
     diff (rel = +1.07e-1, abs = +1.91e-8), tol (rel = +1.00e-2, abs = +7.52e-37)
4177271 more errors...
``

 
**To Reproduce**

Here are my modifs:

```diff
--- a/examples/text-generation/Cargo.toml
+++ b/examples/text-generation/Cargo.toml
@@ -12,7 +12,8 @@ f16 = []
                                                                                                                                      
 [dependencies]
 # Burn
-burn = {path = "../../crates/burn", features=["train", "tch"]}
+burn = {path = "../../crates/burn", features=["train", "wgpu"]}
+# burn = {path = "../../crates/burn", features=["train", "tch"]}
```

```diff
--- a/examples/text-generation/examples/text-generation.rs
+++ b/examples/text-generation/examples/text-generation.rs
@@ -1,4 +1,6 @@
-use burn::optim::decay::WeightDecayConfig;
+#![recursion_limit = "256"]
+
+use burn::{backend::wgpu::WgpuDevice, optim::decay::WeightDecayConfig};
 use text_generation::{DbPediaDataset, training::ExperimentConfig};
                                                                                                                                      
 #[cfg(feature = "f16")]
@@ -6,7 +8,8 @@ type Elem = burn::tensor::f16;
 #[cfg(not(feature = "f16"))]
 type Elem = f32;
                                                                                                                                      
-type Backend = burn::backend::Autodiff<burn::backend::LibTorch<Elem>>;
+// type Backend = burn::backend::Autodiff<burn::backend::LibTorch<Elem>>;
+type Backend = burn::backend::Autodiff<burn::backend::wgpu::Wgpu<Elem>>;
                                                                                                                                      
 fn main() {
     let config = ExperimentConfig::new(
@@ -16,11 +19,14 @@ fn main() {
     );
                                                                                                                                      
     text_generation::training::train::<Backend, DbPediaDataset>(
-        if cfg!(target_os = "macos") {
-            burn::tensor::Device::<Backend>::Mps
-        } else {
-            burn::tensor::Device::<Backend>::Cuda(0)
-        },
+        // if cfg!(target_os = "macos") {
+        //     burn::tensor::Device::<Backend>::Mps
+        // } else {
+        //     burn::tensor::Device::<Backend>::Cuda(0)
+        // },
+        // WgpuDevice::Cpu,
+        WgpuDevice::IntegratedGpu(0),
+        // WgpuDevice::default(),
         DbPediaDataset::train(),
         DbPediaDataset::test(),
         config,

```

Then `cargo run -r --example text-generation`

**Expected behavior**

It should be working (even if it is likely slower).

Content of /tmp/text-generation logs

```
2025-05-21T03:25:44.583314Z  WARN wgpu_hal::vulkan::instance: Unable to find extension: VK_EXT_physical_device_drm    
2025-05-21T03:25:44.598849Z  INFO cubecl_wgpu::runtime: Using adapter AdapterInfo { name: "AMD Radeon Graphics (RADV RENOIR)", vendor: 4098, device: 5686, device_type: IntegratedGpu, driver: "radv", driver_info: "Mesa 24.2.8-1ubuntu1~24.04.1", backend: Vulkan }    
2025-05-21T03:25:44.604372Z  INFO cubecl_wgpu::runtime: Created wgpu compute server on device Device { inner: Core(CoreDevice { context: ContextWgpuCore { type: "Native" }, id: Id(0,1), error_sink: Mutex { data: ErrorSink }, features: Features { features_wgpu: FeaturesWGPU(TEXTURE_FORMAT_16BIT_NORM | TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PIPELINE_STATISTICS_QUERY | TIMESTAMP_QUERY_INSIDE_ENCODERS | TIMESTAMP_QUERY_INSIDE_PASSES | MAPPABLE_PRIMARY_BUFFERS | TEXTURE_BINDING_ARRAY | BUFFER_BINDING_ARRAY | STORAGE_RESOURCE_BINDING_ARRAY | SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING | STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING | PARTIALLY_BOUND_BINDING_ARRAY | MULTI_DRAW_INDIRECT | MULTI_DRAW_INDIRECT_COUNT | PUSH_CONSTANTS | ADDRESS_MODE_CLAMP_TO_ZERO | ADDRESS_MODE_CLAMP_TO_BORDER | POLYGON_MODE_LINE | POLYGON_MODE_POINT | CONSERVATIVE_RASTERIZATION | VERTEX_WRITABLE_STORAGE | CLEAR_TEXTURE | SPIRV_SHADER_PASSTHROUGH | MULTIVIEW | TEXTURE_ATOMIC | TEXTURE_FORMAT_NV12 | SHADER_F64 | SHADER_I16 | SHADER_PRIMITIVE_INDEX | SHADER_INT64 | SUBGROUP | SUBGROUP_VERTEX | SUBGROUP_BARRIER | PIPELINE_CACHE | SHADER_INT64_ATOMIC_MIN_MAX | SHADER_INT64_ATOMIC_ALL_OPS | TEXTURE_INT64_ATOMIC), features_webgpu: FeaturesWebGPU(DEPTH_CLIP_CONTROL | DEPTH32FLOAT_STENCIL8 | TEXTURE_COMPRESSION_BC | TEXTURE_COMPRESSION_BC_SLICED_3D | TIMESTAMP_QUERY | INDIRECT_FIRST_INSTANCE | SHADER_F16 | RG11B10UFLOAT_RENDERABLE | BGRA8UNORM_STORAGE | FLOAT32_FILTERABLE | DUAL_SOURCE_BLENDING) } }) } => AdapterInfo { name: "AMD Radeon Graphics (RADV RENOIR)", vendor: 4098, device: 5686, device_type: IntegratedGpu, driver: "radv", driver_info: "Mesa 24.2.8-1ubuntu1~24.04.1", backend: Vulkan }    
2025-05-21T03:25:44.571825Z  INFO burn_train::learner::train_val: Fitting the model:
 TextGenerationModel {
  transformer: TransformerEncoder {d_model: 384, d_ff: 1536, n_heads: 12, n_layers: 6, dropout: 0.1, norm_first: true, quiet_softmax: false, params: 10646784}
  embedding_token: Embedding {n_embedding: 50260, d_model: 384, params: 19299840}
  embedding_pos: Embedding {n_embedding: 512, d_model: 384, params: 196608}
  output: Linear {d_input: 384, d_output: 50260, bias: true, params: 19350100}
  vocab_size: 50260
  pad_token: 50259
  max_seq_length: 512
  params: 49493332
}    
2025-05-21T03:25:44.764939Z  INFO burn_train::learner::epoch: Executing training step for epoch 1    
2025-05-21T03:25:44.775001Z  INFO burn_train::learner::epoch: Iteration 1    
2025-05-21T03:25:44.794983Z  INFO cubecl_runtime::tune::tune_cache: Load autotune cache ...    
2025-05-21T03:25:44.795026Z  INFO cubecl_runtime::tune::tune_cache: Loaded 13 autotune cached entries    
2025-05-21T03:25:44.800605Z  INFO cubecl_runtime::tune::tune_cache: Load autotune cache ...    
2025-05-21T03:25:44.800646Z  INFO cubecl_runtime::tune::tune_cache: Loaded 16 autotune cached entries    
2025-05-21T03:25:44.856585Z  INFO cubecl_runtime::tune::tune_cache: Load autotune cache ...    
2025-05-21T03:25:44.856653Z  INFO cubecl_runtime::tune::tune_cache: Loaded 16 autotune cached entries    
2025-05-21T03:25:44.865606Z  INFO cubecl_runtime::tune::tune_cache: Load autotune cache ...    
2025-05-21T03:25:44.865648Z  INFO cubecl_runtime::tune::tune_cache: Loaded 3 autotune cached entries    
2025-05-21T03:25:44.888243Z  INFO cubecl_runtime::tune::tuner: Tuning FusedReduceAutotuneKey - ReduceKey: ReduceAutotuneKey { elem_input: Float(F32), elem_output: Float(F32), elem_acc: Float(F32), potential_line_size: 1, axis_is_contiguous: true, reduce_axis_shape: 128, reduce_count: 16384 }, FuseNumReads: 2, FuseNumWrites: 4, FuseNumOps: 1    
2025-05-21T03:25:44.888890Z  INFO cubecl_runtime::tune::tuner: Tuning ReduceAutotuneKey - ElemInput: Float(F32), ElemOutput: Float(F32), ElemAcc: Float(F32), PotentialLineSize: 1, AxisIsContiguous: true, ReduceAxisShape: 128, ReduceCount: 16384    
2025-05-21T03:25:45.140880Z  INFO cubecl_runtime::tune::tuner: Tuning FusedReduceAutotuneKey - ReduceKey: ReduceAutotuneKey { elem_input: Float(F32), elem_output: Float(F32), elem_acc: Float(F32), potential_line_size: 1, axis_is_contiguous: true, reduce_axis_shape: 128, reduce_count: 16384 }, FuseNumReads: 2, FuseNumWrites: 4, FuseNumOps: 2    
2025-05-21T03:25:48.740371Z  INFO cubecl_runtime::tune::tune_cache: Load autotune cache ...    
2025-05-21T03:25:48.740412Z  INFO cubecl_runtime::tune::tune_cache: Loaded 6 autotune cached entries    
2025-05-21T03:25:48.740420Z  INFO cubecl_runtime::tune::tuner: Tuning SumAutotuneKey { dtype: F32, length: 558 }    
2025-05-21T03:25:48.955285Z  INFO cubecl_runtime::tune::tuner: Tuning FusedReduceAutotuneKey - ReduceKey: ReduceAutotuneKey { elem_input: Float(F32), elem_output: Float(F32), elem_acc: Float(F32), potential_line_size: 2, axis_is_contiguous: false, reduce_axis_shape: 16, reduce_count: 16384 }, FuseNumReads: 1, FuseNumWrites: 2, FuseNumOps: 0    
2025-05-21T03:25:55.811810Z ERROR burn_train::learner::application_logger: PANIC => panicked at /home/johann/projects/burn/crates/burn-cubecl-fusion/src/shared/trace/base.rs:94:38:
Tensors are not approx eq:
  => Position 1: 0.00000007623079 != 0.000000104531644
     diff (rel = +2.71e-1, abs = +2.83e-8), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 2: 0.00000018476716 != 0.00000006408816
     diff (rel = +6.53e-1, abs = +1.21e-7), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 3: 0.00000018888448 != 0.00000030326353
     diff (rel = +3.77e-1, abs = +1.14e-7), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 4: 0.00000014465004 != 0.00000035931777
     diff (rel = +5.97e-1, abs = +2.15e-7), tol (rel = +1.00e-2, abs = +7.52e-37)
  => Position 5: 0.00000007623079 != 0.000000043319783
     diff (rel = +4.32e-1, abs = +3.29e-8), tol (rel = +1.00e-2, abs = +7.52e-37)
4623904 more errors...    

```

**Comment by nathanielsimard** [comment]
The examples should not be executed from the top level repo because some testing features flags are activated inside the workspace, not sure how to fix the usability issue here.

**Comment by tafia** [comment]
Oh nice! Thank you!
